### PR TITLE
Feature/report failing driver commands

### PR DESCRIPTION
**Pull request by basdijkstra**
This PR offers more customization around driver command reporting, letting the user choose between reporting all driver commands, none, or only the failing ones.

### CODE REVIEW

File: TestProject.OpenSDK/Enums/DriverCommands.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        /// <summary>
+        /// Only report failing driver commands.
+        /// </summary>
+        Passing,
+
+        /// <summary>
+        /// Never report driver commands.
+        /// </summary>
+        All,
```
Comment (mstrelex) [inline TestProject.OpenSDK/Enums/DriverCommands.cs:37]: It's probably nothing, but the comma is here on purpose?

File: TestProject.OpenSDK/Enums/DriverCommands.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace TestProject.OpenSDK.Enums
+{
+    /// <summary>
+    /// Enumeration of options for driver command reporting.
+    /// </summary>
+    public enum DriverCommands
```
Comment (mstrelex) [inline TestProject.OpenSDK/Enums/DriverCommands.cs:22]: Not sure about this name... Maybe `DriverCommandsFilter`?

File: TestProject.OpenSDK/Internal/Helpers/CommandExecutors/ReportingCommandExecutor.cs [hunk 3]
```diff
@@ -194,25 +195,41 @@ public string GetScreenshot()
...
+                {
+                    command = RedactHelper.RedactCommand(this.commandExecutor, command);
+                }
 
-            AgentClient.GetInstance().ReportDriverCommand(driverCommandReport);
+                DriverCommandReport driverCommandReport = new DriverCommandReport(command.Name, command.Parameters, result, passed);
+
+                if (!passed)
+                {
+                    driverCommandReport.Screenshot = this.GetScreenshot();
```
Comment (mstrelex) [inline TestProject.OpenSDK/Internal/Helpers/CommandExecutors/ReportingCommandExecutor.cs:228]: Don't we have yet the option to control screenshots for the whole test as we do in Python?

File: TestProject.OpenSDK.Tests/Examples/Reports/ManualReportingTest.cs [hunk 4]
```diff
@@ -40,6 +41,9 @@ public void StartBrowser()
             this.driver = new ChromeDriver(
                 projectName: "Examples",
                 jobName: "Manual reporting examples");
+
+            this.driver.Report().DisableAutoTestReports(true);
+            this.driver.Report().DisableCommandReports(DriverCommands.All);
```
Comment (mstrelex) [inline TestProject.OpenSDK.Tests/Examples/Reports/ManualReportingTest.cs:46]: Perhaps a comment here would be nice to explain what exactly is expected to happen.

**Review by mstrelex — CHANGES_REQUESTED** [review]

File: TestProject.OpenSDK/Enums/DriverCommands.cs (hunk 1 above)
Comment (basdijkstra) [reply TestProject.OpenSDK/Enums/DriverCommands.cs:37]: Yes. The build fails when it's not there (.NET analyzer rule).

File: TestProject.OpenSDK/Enums/DriverCommands.cs (hunk 2 above)
Comment (basdijkstra) [reply TestProject.OpenSDK/Enums/DriverCommands.cs:22]: I like that. I've thought long about a good name but this is better than what I came up with :)

File: TestProject.OpenSDK/Internal/Helpers/CommandExecutors/ReportingCommandExecutor.cs (hunk 3 above)
Comment (basdijkstra) [reply TestProject.OpenSDK/Internal/Helpers/CommandExecutors/ReportingCommandExecutor.cs:228]: No, not yet at least.

File: TestProject.OpenSDK.Tests/Examples/Reports/ManualReportingTest.cs (hunk 4 above)
Comment (basdijkstra) [reply TestProject.OpenSDK.Tests/Examples/Reports/ManualReportingTest.cs:46]: That's a good idea.

**Pull request merged**